OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -69,26 +69,15 @@ public override bool Tick(Actor self)
 				return true;
 
 			if (IsCanceling)
-			{
-				if (carryall.State == Carryall.CarryallState.Reserved)
-					carryall.UnreserveCarryable(self);
-
-				// Make sure we run the TakeOff activity if we are / have landed
-				if (self.Trait<Aircraft>().HasInfluence())
-				{
-					ChildHasPriority = true;
-					IsInterruptible = false;
-					QueueChild(new TakeOff(self));
-					return false;
-				}
-
 				return true;
-			}
 
 			if (cargo.IsDead || carryable.IsTraitDisabled || !cargo.AppearsFriendlyTo(self))
 			{
+				// When we cancel the activity we keep the carryable reserved if we had already picked it up
+				// However if the carryable becomes invalid, we want to unreserve it in all cases
```

Comment:
Seeing how quickly you removed the if check, the other location probably needs a comment too.

Reply:
Moved the unreserving to FerryUnit activity. The other unreserving call on cancel remains as a way for FerryUnit to realise that PickupUnit was cancelled by something other than the unit dieing